src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java

```diff
@@ -181,15 +181,18 @@ public static DetailAST appendHiddenCommentNodes(DetailAST root) {
             }
             curNode = toVisit;
         }
-        if (lastNode != null && ((DetailAstImpl) lastNode).getHiddenAfter() != null) {
-            DetailAST currentSibling = lastNode;
-            for (Token token: ((DetailAstImpl) lastNode).getHiddenAfter()) {
-                final DetailAST newCommentNode =
-                        createCommentAstFromToken((CommonToken) token);
+        if (lastNode != null) {
+            final List<Token> hiddenAfter = ((DetailAstImpl) lastNode).getHiddenAfter();
+            if (hiddenAfter != null) {
+                DetailAST currentSibling = lastNode;
+                for (Token token : hiddenAfter) {
```

Comment:
Not a big fan of this change, but I really can't think of a better way to do this without a bunch of other refactoring.

Reply:
Resolved